src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -17,13 +17,31 @@
 import frc.robot.drive.MotorSpeedController;
 
 public class Drivetrain extends SubsystemBase {
-    MotorSpeedController motor1 = new MotorSpeedController(Constants.DRIVE_MOTOR_1_ID, Constants.DRIVE_MOTOR_1_TYPE);
-    MotorSpeedController motor2 = new MotorSpeedController(Constants.DRIVE_MOTOR_2_ID, Constants.DRIVE_MOTOR_2_TYPE);
+    SpeedControllerGroup leftMotorGroup;
+    MotorSpeedController leftMotorLeader;
+    MotorSpeedController leftMotorFollower = null;
+
+    SpeedControllerGroup rightMotorGroup;
+    MotorSpeedController rightMotorLeader;
+    MotorSpeedController rightMotorFollower = null;
     
-    DifferentialDrive diffDrive = new DifferentialDrive(motor1, motor2);
+    DifferentialDrive diffDrive;
 
     public Drivetrain() {
         super();
+
+        leftMotorLeader = new MotorSpeedController(Constants.DRIVE_MOTOR_LEFT_LEADER_ID, Constants.DRIVE_MOTOR_LEFT_LEADER_TYPE);
+        rightMotorLeader = new MotorSpeedController(Constants.DRIVE_MOTOR_RIGHT_LEADER_ID, Constants.DRIVE_MOTOR_RIGHT_LEADER_TYPE);
+
+        if (Constants.DRIVE_DUAL_MOTORS) {
+            leftMotorFollower = new MotorSpeedController(Constants.DRIVE_MOTOR_LEFT_FOLLOWER_ID, Constants.DRIVE_MOTOR_LEFT_FOLLOWER_TYPE);
+            rightMotorFollower = new MotorSpeedController(Constants.DRIVE_MOTOR_RIGHT_FOLLOWER_ID, Constants.DRIVE_MOTOR_RIGHT_FOLLOWER_TYPE);
+        }
+
+        leftMotorGroup = new SpeedControllerGroup(leftMotorLeader, leftMotorFollower);
+        rightMotorGroup = new SpeedControllerGroup(rightMotorLeader, rightMotorFollower);
```

Comment:
Does this work if the followers are null?

Reply:
No, but we made the change in #2, should we also make the change in this branch?

Reply:
I'll just make the change since I am fixing other parts of drivetrain